### PR TITLE
Add new client to get register data

### PR DESCRIPTION
**Pull request by sophiebrown01**
This PR adds functionality to allow clients to get the records, entries and metadata of registers without the need to make repeated API calls. In particular, it can deliver all the records of a register, only the current / expired / current & expired records of a register, and individual pieces of metadata such as the field and register definitions.

### CODE REVIEW

File: lib/register_client.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    def get_current_records
+      get_records.select { |record| record[:item]['end-date'].blank? }
+    end
+
+    def get_expired_records
+      get_records.select { |record| record[:item]['end-date'].present? }
+    end
+
+    def get_register_data(register, phase)
```
Comment (gidsg) [inline lib/register_client.rb]: I think this is not used now?

File: lib/register_client.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      refresh_data
+    end
+
+    def refresh_data
+      rsf = download_rsf(@register, @phase)
+      parse_rsf rsf
+    end
+
+    def download_rsf(register, phase)
+      RestClient.get('https://' + register + '.' + phase + '.openregister.org/download-rsf')
```
Comment (gidsg) [inline lib/register_client.rb]: should we use [string interpolation](https://stackoverflow.com/documentation/ruby/834/strings/2819/difference-between-single-quoted-and-double-quoted-string-literals#t=201709131624075680173) for this?

**Comment by gidsg** [comment]
Is it worth documenting the new methods in the Readme, or that will that make things more confusing while we have multiple approaches in the library?

File: lib/openregister.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 require 'morph'
 require 'rest-client'
+require 'json'
+require 'registers_client'
+require 'register_client'
 
 module OpenRegister
   VERSION = '0.0.1' unless defined? OpenRegister::VERSION
```
Comment (gidsg) [inline lib/openregister.rb]: should we bump the version number and tag a release?

Comment (sophiebrown01) [reply lib/openregister.rb]: I was going to bump the version number. Not sure about tagging a release. Also, the version is defined in the `gemspec` file I believe.

File: lib/register_client.rb (hunk 2 above)
Comment (sophiebrown01) [reply lib/register_client.rb]: Let's do it.

**Comment by sophiebrown01** [comment]
Regarding documentation: until we've fleshed out the client library a bit more in terms of the upcoming work we have on the front-end, it might be worth holding off on updating the README?

File: lib/register_client.rb (hunk 1 above)
Comment (sophiebrown01) [reply lib/register_client.rb]: Removing.

File: lib/register_client.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
+require 'rest-client'
+require 'json'
+
+module OpenRegister
+  class RegisterClient
+    def initialize register, phase
```
Comment (dankmitchell) [inline lib/register_client.rb]: Add parenthese

File: lib/register_client.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      @items = []
+      @entries = { user: [], system: [] }
+      @records = { user: {}, system: {} }
+
+      refresh_data
+    end
+
+    def refresh_data
+      rsf = download_rsf(@register, @phase)
+      parse_rsf rsf
```
Comment (dankmitchell) [inline lib/register_client.rb]: Add parenthese

**Review by dankmitchell — APPROVED** [review]

**Pull request merged**